repos/system_upgrade/common/actors/scansubscriptionmanagerinfo/tests/test_scansubscriptionmanagementinfo.py

```diff
@@ -0,0 +1,349 @@
+'''
+
+RHSMRepositoryEntry = namedtuple('RHSMRepositoryEntry', ('id', 'name', 'url', 'enabled'))  # For clarity purposes
+RHSM_ENABLED_REPOS = [
+    RHSMRepositoryEntry(
+        id='rhel-8-for-x86_64-appstream-rpms',
+        name='Appstream',
+        url='some_url',
+        enabled='1'),
+    RHSMRepositoryEntry(
+        id='satellite-tools-6.6-for-rhel-8-x86_64-rpms',
+        name='Satellite',
+        url='some_url',
+        enabled='1'),
+    RHSMRepositoryEntry(
+        id='rhel-8-for-x86_64-baseos-rpms',
+        name='Base',
+        url='some_url',
+        enabled='1')
+]
+
+
+class MockedCommandRunner(object):
+    """Mock for the api.run that allows mapping of called commands to desired output."""
+    def __init__(self):
+        self.mocked_command_calls = dict()  # A map from called commands to their mocked output
+        self.calls_performed_counter = 0
+
+    def add_mocked_command_with_stdout(self, command, stdout):
+        self.mocked_command_calls[command] = {'stdout': stdout}
+
+    def __call__(self, command, **kwargs):
+        # Ignore kwargs
+        default_result = {'stdout': ''}
+        self.calls_performed_counter += 1
+        return self.mocked_command_calls.get(tuple(command), default_result)
+
+
+def mocked_get_parsed_repofiles(context):
+    return []
+
+
+CommonMonkeypatchedAttributes = namedtuple('CommonMultipatchedAttributes',
+                                           ['actor', 'actor_products', 'external_command_runner'])
+
+
+@pytest.fixture
+def common_monkeypatched_handles(monkeypatch):
+    """Peforms monkeypatching commonly done in the tests and returns the handles to the used monkeypatched objects."""
+    runner = MockedCommandRunner()
+    actor_products = produce_mocked()
+    actor = CurrentActorMocked()
+
+    # Patch the repofileutils so that the tests can pass without the `dnf` module
+    monkeypatch.setattr(rhsm.repofileutils, 'get_parsed_repofiles', mocked_get_parsed_repofiles)
+
+    monkeypatch.setattr(mounting, 'run', runner)
+    monkeypatch.setattr(api, 'current_actor', actor)
+    monkeypatch.setattr(api, 'produce', actor_products)
+
+    return CommonMonkeypatchedAttributes(actor=actor, actor_products=actor_products, external_command_runner=runner)
+
+
+def test_scansubscriptionmanagementinfo_sku_listing(monkeypatch, common_monkeypatched_handles):
+    """Tests whether the actor can list the SKUs correctly."""
+
+    _, actor_products, command_runner = common_monkeypatched_handles
+
+    command_runner.add_mocked_command_with_stdout(RHSM_LIST_CONSUMED, 'SKU: 598339696910')
+
+    scanrhsm.scan()
+    assert command_runner.calls_performed_counter > 0, 'Some calls to subscription-manager were expected.'
+    assert actor_products.model_instances, 'The actor did not produce a message.'
+
+    produced_message = actor_products.model_instances[0]
+    assert produced_message.attached_skus, 'Actor failed to parse SKU.'
+    assert len(produced_message.attached_skus) == 1, \
```

Comment:
Do not use `\` to split/wrap....(you know)... lines.

Reply:
New version does not contain any usage of `\` to achieve multi-line asserts.